daemon/daemon.go

```diff
@@ -445,7 +445,23 @@ func (daemon *Daemon) DaemonLeavesCluster() {
 	// Daemon is in charge of removing the attachable networks with
 	// connected containers when the node leaves the swarm
 	daemon.clearAttachableNetworks()
+	// We no longer need the cluster provider, stop it now so that
+	// the network agent will stop listening to cluster events.
```

Comment:
Please refer to https://github.com/docker/docker/pull/30157#discussion_r98034445 where the function naming was decided.

This function takes care of the things that need to be done when the node leaves the cluster.
At the same time the same set of actions needs to be taken care when a node is shutting down.

I did not want to duplicate the same code for the shutdown path.